pombola/kenya/views_facebook_experiments.py

```diff
@@ -0,0 +1,200 @@
+        self.survey_session = request.GET.get('sid')
+
+        if self.survey_session:
+            recordPositiveResponse(self.survey_session, '61', '10172')
+
+        return super(TreatmentPolitics, self).get(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super(TreatmentPolitics, self).get_context_data(**kwargs)
+        context['form_counties'] = ConstituencyGroupedByCountySelectForm()
+        context['sid'] = self.survey_session
+        return context
+
+
+class TreatmentPoliticsThanks(TemplateView):
+
+    template_name = 'facebook-experiment-politics-thanks.html'
+
+    def get(self, request, *args, **kwargs):
+        survey_session = request.GET.get('sid')
+
+        if survey_session:
+            recordPositiveResponse(survey_session, '59', '10171')
+
+        return super(TreatmentPoliticsThanks, self).get(request, *args, **kwargs)
+
+
+class SurveyGizmoRecord(TemplateView):
+
+    # The survey doesn't actually care if this works or not, just send an "OK".
+    template_name = 'facebook-ajax-response.html'
+
+    def get(self, request, *args, **kwargs):
+        survey_session = request.GET.get('sid')
+        question_id = request.GET.get('qid')
+        selection_id = request.GET.get('selid')
+
+        if survey_session and question_id and selection_id:
+            recordPositiveResponse(survey_session, question_id, selection_id)
+
+        return super(SurveyGizmoRecord, self).get(request, *args, **kwargs)
+
+
+class SurveyGizmoValue(TemplateView):
+
+    # The survey doesn't actually care if this works or not, just send an "OK".
+    template_name = 'facebook-ajax-response.html'
+
+    def get(self, request, *args, **kwargs):
+        survey_session = request.GET.get('sid')
+        question_id = request.GET.get('qid')
+        value = request.GET.get('value')
+
+        if survey_session and question_id and value:
+            recordResponseValue(survey_session, question_id, value)
+
+        return super(SurveyGizmoValue, self).get(request, *args, **kwargs)
+
+
+def getSurveyIdFromSession(survey_session):
+
+    survey_id = cache.get(survey_session)
+
+    if not survey_id:
+
+        url = "https://restapi.surveygizmo.com/v4/survey/2802241/surveyresponse/"
+
+        querystring = {
+            "api_token": settings.SURVEYGIZMO_API_TOKEN,
+            "api_token_secret": settings.SURVEYGIZMO_API_SECRET,
+            "filter[field][0]": "[question(13)]",
+            "filter[operator][0]": "=",
+            "filter[value][0]": survey_session
+            }
+
+        headers = {
+            'cache-control': "no-cache"
+            }
+
+        response = requests.request("GET", url,
+                                    headers=headers,
+                                    params=querystring
+                                    )
+        data = json.loads(response.text)
+
+        # Sanity check we only have one response. If not, abandon ship.
```

Comment:
If the authentication details are missing or wrong, then `data` won't contain a `total_count` key - it'll be something like:

```
{u'message': u'Login failed / Invalid auth token', u'code': 401, u'result_ok': False}
```

Rather than throwing a `KeyError` in this case, it would be better to throw an `Exception` with a message that indicates what's wrong.